src/shell.html

```diff
@@ -1220,30 +1220,34 @@
       var statusElement = document.getElementById('status');
       var progressElement = document.getElementById('progress');
       var spinnerElement = document.getElementById('spinner');
+      var writeToConsoleElement = (function() {
+        var element = document.getElementById('output');
+        if (element) element.value = ''; // clear browser cache
+        return function(text) {
+          // These replacements are necessary if you render to raw HTML
+          //text = text.replace(/&/g, "&amp;");
+          //text = text.replace(/</g, "&lt;");
+          //text = text.replace(/>/g, "&gt;");
+          //text = text.replace('\n', '<br>', 'g');
+          if (element) {
+            element.value += text + "\n";
+            element.scrollTop = element.scrollHeight; // focus on bottom
+          }
+        };
+      })();
 
       var Module = {
         preRun: [],
         postRun: [],
-        print: (function() {
-          var element = document.getElementById('output');
-          if (element) element.value = ''; // clear browser cache
-          return function(text) {
-            if (arguments.length > 1) text = Array.prototype.slice.call(arguments).join(' ');
-            // These replacements are necessary if you render to raw HTML
-            //text = text.replace(/&/g, "&amp;");
-            //text = text.replace(/</g, "&lt;");
-            //text = text.replace(/>/g, "&gt;");
-            //text = text.replace('\n', '<br>', 'g');
-            console.log(text);
-            if (element) {
-              element.value += text + "\n";
-              element.scrollTop = element.scrollHeight; // focus on bottom
-            }
-          };
-        })(),
+        print: function(text) {
+          if (arguments.length > 1) text = Array.prototype.slice.call(arguments).join(' ');
+          console.log(text);
+          writeToConsoleElement(text);
+        },
         printErr: function(text) {
           if (arguments.length > 1) text = Array.prototype.slice.call(arguments).join(' ');
           console.error(text);
+          writeToConsoleElement(text);
```

Comment:
I think I understand your proposal, but I think there are also reasonable reasons to keep things as they are. Adding more logging for users might surprise them, so I'd rather not inconvenience them unless we are sure we want to do this. For example, a developer's production build might have stderr logging that now suddenly shows up in front of end users of their website.

Perhaps a poll among our users could make sense as I suggested, what do you think?

Reply:
But this change only effect shell.html.. unless folks are using that unmodified I don't see how this would effect anything in production.

Reply:
I'm happy to re-think this BTW, no hurry the land it.

Reply:
Hmm, that's a fair point, likely most users have their own custom shell.html. Maybe that does shift things towards your point of view.

I do still feel hesitant at making a change in something so user-visible, even if it is only users in test mode, given the tradeoff is not obvious here. But I don't feel strongly. I would at least like to get some more opinions on this, though.

Reply:
Discussing filesystem topics with @tlively and @ethanalee , I realized something relevant for this PR. Currently we proxy all filesystem operations to the main thread anyhow, but we want to do a lot less of that. In the future, it would be nice for stdout to be proxied to the main thread - so that it can be shown on the HTML - but for stderr to *not* be proxied, and just sent to `console.error` immediately. This PR would show stderr on the HTML, so we'd need to proxy stderr as well.

(Maybe not a huge deal, as a high-performance application shouldn't be printing lots of stderr.)

Reply:
I think whether of not stdout (or stderr) is displayed at all or if it goes to HTML or to a file is up the application.  By default I think we probably want to just use `console` for both,  but for the `shell.html` we probably want to proxy both back for display to the user.

For most production use case the user probably doesn't want to too see either stdout *or* stderr, but for a developer looking an html console I would guess they want to see both.